stl/inc/regex

```diff
@@ -1998,23 +1989,24 @@ private:
     }
 
     template <class _InIt>
-    void _Reset(_InIt _First, _InIt _Last, flag_type _Flags, input_iterator_tag) {
-        // build regular expression from input iterators
-        basic_string<_Iter_value_t<_InIt>> _Str(_First, _Last);
-
-        _Reset(_Str.data(), _Str.data() + static_cast<ptrdiff_t>(_Str.size()), _Flags, forward_iterator_tag{});
-    }
-
-    template <class _FwdIt>
-    void _Reset(_FwdIt _First, _FwdIt _Last, flag_type _Flags, forward_iterator_tag) {
-        // build regular expression from forward iterators
+    void _Reset(_InIt _First, _InIt _Last, flag_type _Flags) {
+        if constexpr (_Is_fwd_iter_v<_InIt>) {
+            // build regular expression from forward iterators
 #if _ENHANCED_REGEX_VISUALIZER
-        _Visualization.assign(_First, _Last);
+            _Visualization.assign(_First, _Last);
 #endif // _ENHANCED_REGEX_VISUALIZER
 
-        _Parser<_FwdIt, _Elem, _RxTraits> _Prs(_Traits, _First, _Last, _Flags);
-        _Root_node* _Rx = _Prs._Compile();
-        _Reset(_Rx);
+            _Parser<_InIt, _Elem, _RxTraits> _Prs(_Traits, _First, _Last, _Flags);
+            _Root_node* _Rx = _Prs._Compile();
+            _Reset(_Rx);
+        } else {
+            static_assert(_Is_input_iter_v<_InIt>, "Should be at least input iterator");

```

Comment:
"Iterators must be input iterators" would be more consistent with the other PRs.